src/doc/tarpl/drop-flags.md

```diff
@@ -0,0 +1,81 @@
+% Drop Flags
+
+The examples in the previous section introduce an interesting problem for Rust.
+We have seen that's possible to conditionally initialize, deinitialize, and
+*reinitialize* locations of memory totally safely. For Copy types, this isn't
+particularly notable since they're just a random pile of bits. However types
+with destructors are a different story: Rust needs to know whether to call a
+destructor whenever a variable is assigned to, or a variable goes out of scope.
```

Comment:
nit: more than just variables being assigned to, it more generally needs to call the destructor in the case when any lvalue of that type-with-a-dtor is overwritten. e.g. `l[val] = expr;` or `l.v.a.l = expr;`